services/platform/platform.go

```diff
@@ -169,7 +169,7 @@ func (s *Service) Disconnect(ctx context.Context, req *platformpb.DisconnectRequ
 	err = s.disconnect(ctx, &disconnectPMMParams{
 		PMMServerID: settings.PMMServerID,
 	})
-	if err != nil {
+	if err != nil && !req.Force {
```

Comment:
probably it should be `|| req.Force`?

Reply:
I don't think so. This block restores the above deleted data.
In case of an error we need to restore it only when the request is not forced.
In case of `|| req.Force` the data will be restored even when there was no error at all, which is wrong. 

I agree it's hard to follow, but it's how the method is designed.

Reply:
I suggest to use transaction in `disconnect`, so it won't be so hard to follow.

Reply:
oh... I see. but that is super unclear. How about a bit of refactoring?

we can have:
1: extra variable:
```
recover := err != nil && !req.Force
if recover {
...
}
```

2: extract method:
```
if err != nil && !req.Force {
 recoverXyz() // here we need a good name, as it the scope grows
}
```

Reply:
Please check usage of transaction here https://github.com/percona/pmm-managed/blob/main/services/management/mysql.go#L58
It rollbacks changes in case of error.

Reply:
ok, sounds reasonable.